.github/workflows/build-check_x86_64.yml

```diff
@@ -9,46 +9,39 @@ jobs:
     steps:
-      - name: Checkout code
-        uses: actions/checkout@v4
-        with:
-          ref: "${{ github.event.pull_request.head.sha }}"
-          fetch-depth: 0
-          path: kernel-src-tree
-
-      - name: Install rinse
-        run: |
-          sudo apt-get install rinse
-
-      - name: Build centos7 chroot
-        run: |
-          sudo rinse --distribution centos-7 \
-                --mirror http://dl.rockylinux.org/vault/centos/7/os/x86_64/Packages \
-                --arch amd64 \
-                --directory centos-7-chroot
-
-      - name: Point yum to vault (in chroot)
+      - name: Point yum to vault
         run: |
-          sudo sed -e '/mirrorlist=.*/d' \
+          sed -e '/mirrorlist=.*/d' \
               -e 's/#baseurl=/baseurl=/' \
               -e "s/\$releasever/7.9.2009/g" \
               -e "s/mirror.centos.org/dl.rockylinux.org\/vault/g" \
-              -i centos-7-chroot/etc/yum.repos.d/CentOS-Base.repo
+              -i /etc/yum.repos.d/CentOS-Base.repo
 
       - name: Install tools and Libraries (in chroot)
         run: |
-          sudo chroot centos-7-chroot yum groupinstall 'Development Tools' -y
-          sudo chroot centos-7-chroot yum install bc dwarves git glibc-devel hostname kernel-devel mpfr openssl openssl-devel elfutils-libelf-devel -y
+          yum groupinstall 'Development Tools' -y
+          yum install bc dwarves git glibc-devel hostname kernel-devel mpfr openssl openssl-devel elfutils-libelf-devel -y
 
-      - name: Build the Kernel (in chroot)
+      - name: Checkout code
         run: |
-          sudo mv kernel-src-tree centos-7-chroot
-          sudo chroot centos-7-chroot sh -c "cd kernel-src-tree && cp configs/kernel-3.10.0-x86_64.config .config"
-          sudo chroot centos-7-chroot sh -c "cd kernel-src-tree && make olddefconfig"
-          sudo chroot centos-7-chroot sh -c "cd kernel-src-tree && make -j$(nproc)"
+          git clone --branch ${{ github.head_ref }} "https://oauth2:$GITHUB_TOKEN@github.com/ctrliq/kernel-src-tree"
+
+      - name: Build the Kernel
+        working-directory: kernel-src-tree
+        run: |
+          cp configs/kernel-3.10.0-x86_64.config .config
+          make olddefconfig
+          make -j$(nproc)
+
       - name: Check kabi
+        working-directory: kernel-src-tree
         run: |
-          sudo chroot centos-7-chroot sh -c "git clone --branch c7 --single-branch https://git.centos.org/rpms/kernel.git kernel-dist-git"
-          sudo chroot centos-7-chroot sh -c "cd kernel-dist-git && git reset --hard imports/c7/kernel-3.10.0-1160.119.1.el7"
-          sudo chroot centos-7-chroot sh -c "./kernel-dist-git/SOURCES/check-kabi -k ./kernel-dist-git/SOURCES/Module.kabi_x86_64 -s ./kernel-src-tree/Module.symvers"
+          git clone --branch c7 --single-branch https://git.centos.org/rpms/kernel.git kernel-dist-git
+          cd kernel-dist-git
+          git reset --hard imports/c7/kernel-3.10.0-1160.119.1.el7
+          cd ..
```

Comment:
Tags are syntactically equivalent to branches, so these four lines can be simplified into:
```
git clone --branch imports/c7/kernel-3.10.0-1160.119.1.el7 --depth 1 https://git.centos.org/rpms/kernel.git kernel-dist-git
```

And using `--depth 1` instead of `--single-branch` produces an even smaller clone because it won't fetch any history at all. (and `--depth` implies `--single-branch`)

Reply:
Yea, but then you are left in a yucky detached head state.   j/k.  It is better as a one liner.  Changed.  Thanks!